frame/support/procedural/src/pallet/parse/extra_constants.rs

```diff
@@ -102,10 +129,25 @@ impl ExtraConstantsDef {
 				syn::ReturnType::Type(_, type_) => *type_.clone(),
 			};
 
+			// parse metadata_name
+			let mut extra_constant_attrs: Vec<ExtraConstAttr> = helper::take_item_pallet_attrs(method)?;
+
+			if extra_constant_attrs.len() > 1 {
+				let msg = "Invalid attribute in pallet::constant_name, only one attribute is expected";
+				return Err(syn::Error::new(extra_constant_attrs[1].metadata_name.span(), msg))
+			}
+
+			let metadata_name = if extra_constant_attrs.len() == 1 {
+				Some(extra_constant_attrs.pop().unwrap().metadata_name)
+			} else {
+				None
+			};
```

Comment:
Maybe simpler
```suggestion
			let metadata_name = extra_constant_attrs.pop().map(|attr| attr.metadata_name);
```

Reply:
Very cool.  I did not know `map` passed forward `Options` like that.